File: cpp/WCFExampleCPP/WCFExampleCPP/EvalService.cpp
#include "IEvalService.cpp"

using namespace System;
using namespace System::Collections::Generic;
using namespace System::Runtime::Serialization;
using namespace System::ServiceModel;

namespace WCFExampleCPP {

	[ServiceBehavior(InstanceContextMode=InstanceContextMode::Single)]
	public ref class EvalService : IEvalService
	{
	public:
		EvalService()
		{
			m_evals = gcnew List<Eval^>();
		}
		virtual void SubmitEval(Eval^ eval)
		{
			eval->Id = Guid::NewGuid().ToString();
			m_evals->Add(eval);
		}

		virtual List<Eval^>^ GetEvals()
		{
			return m_evals;
		}
		virtual void RemoveEval(String^ id)
		{
			Eval^ foundEval = nullptr;
			int EvalCnt = m_evals->Count;
			for (int currEval = 0; 
				 currEval < EvalCnt;
				 currEval++)
			{
				Eval^ ev = m_evals[currEval];
				if(ev->Id == id)
				{
					Console::WriteLine("FOUNDIT");
					foundEval = ev;
				}
			}
			if( foundEval != nullptr)
			{
				m_evals->Remove(foundEval);
			}
		}

	private:
		List<Eval^>^ m_evals;
	};
}

File: csharp/WCFEvalExample/Backup/WCFEvalExample/EvalService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.ServiceModel;

namespace WCFEvalExample
{
    [ServiceBehavior(InstanceContextMode=InstanceContextMode.Single)]
    public class EvalService : IEvalService
    {
        #region IEvalService Members
        List<Eval> m_evals = new List<Eval>();
        public void SubmitEval(Eval eval)
        {
            eval.Id = Guid.NewGuid().ToString();
            m_evals.Add(eval);
        }

        public List<Eval> GetEvals()
        {
            return m_evals;
        }

        public void RemoveEval(string id)
        {
            m_evals.Remove(m_evals.Find(e => e.Id.Equals(id)));
        }

        #endregion
    }
}
